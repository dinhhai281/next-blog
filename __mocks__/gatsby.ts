import React from 'react';
const gastby = jest.requireActual('gatsby');

module.exports = {
  ...gastby,
  graphql: jest.fn(),
  Link: jest
    .fn()
    .mockImplementation(
      ({ activeClassName, activeStyle, getProps, innerRef, partiallyActive, ref, replace, to, ...rest }) =>
        React.createElement('a', { ...rest, href: to })
    ),
  StaticQuery: jest.fn(),
  useStaticQuery: jest.fn(),
};
