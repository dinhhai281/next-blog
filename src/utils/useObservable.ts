import { useEffect, SetStateAction, Dispatch } from 'react';
import { Observable } from 'rxjs';

export const useObservable = <T extends unknown>(input$: Observable<T>, setter: Dispatch<SetStateAction<T>>) => {
  useEffect(() => {
    const subscription = input$.subscribe(setter);

    return () => subscription.unsubscribe();
  }, [input$]);
};
