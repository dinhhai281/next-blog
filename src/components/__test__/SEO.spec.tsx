import { siteMetadataStub } from '@app/utils/testing';
import { render } from '@testing-library/react';
import { useStaticQuery } from 'gatsby';
import React from 'react';

import { SEO } from '../SEO';

describe('SEO', () => {
  it('should render correctly', () => {
    (useStaticQuery as jest.Mock).mockImplementationOnce(() => ({
      site: { siteMetadata: siteMetadataStub },
    }));

    const { asFragment } = render(<SEO />);
    expect(asFragment()).toMatchSnapshot();
  });
});
