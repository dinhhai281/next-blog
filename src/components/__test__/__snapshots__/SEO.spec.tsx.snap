// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO should render correctly 1`] = `<DocumentFragment />`;
